### DISCUSSION TITLE
Conditional Queueing

### DISCUSSION THREAD

**Discussion by AlexJump24** [spatie/laravel-queueable-action#56] [Ideas]
Hi,

Really enjoying using this package.

One question I have is would you ever consider supporting conditional dispatching e.g. like in regular jobs in laravel with `dispatchIf` and `dispatchUnless`.

Its more of a style over substance thing, however I really love the linear flow of using those commands and is the only thing I felt was missing when using this.

If it was any interest I'm not sure how it would fit into the API though given actions are not dispatched but instead just queued. 

Anyways, thanks for all the work you do, we love your work.

**Comment by brendt** [comment]
We could add support for it, since we already proxy several methods from the queueable action to the job. Feel free to send a tested PR.
